### PR TITLE
[JENKINS-71089] Fix `hasHeader` attribute for `f:hetero-list`

### PR DESCRIPTION
**Pull request by janfaracik**
See [JENKINS-71089](https://issues.jenkins.io/browse/JENKINS-71089).

Restores the `hasHeader = false` functionality.

### Testing done

image

* Changed a hetero-list on `core` (as there isn't an existing one) to set the has header attribute to false, the title/dragdrop handle/help no longer shows now.

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7853"></a>

### CODE REVIEW

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by daniel-beck — APPROVED** [review]
<details>
Tested a PR build with the same code I discovered the issue with originally.

> Screenshot

**Comment by NotMyFault** [comment]
Caused [JENKINS-71156](https://issues.jenkins.io/browse/JENKINS-71156)

**Comment by MarkEWaite** [comment]
@janfaracik I see your 👍 on the comment from @NotMyFault , but I don't see a pull request that resolves [JENKINS-71156](https://issues.jenkins.io/browse/JENKINS-71156).  Is there a fix in progress or should we consider reverting this change?

The report from @daniel-beck in [JENKINS-71089](https://issues.jenkins.io/browse/JENKINS-71089) did not seem to be as readily detected as [JENKINS-71156](https://issues.jenkins.io/browse/JENKINS-71156).  The behavior in JENKINS-71089 has existed since 2.335, while the behavior reported in JENKINS-71156 has only been seen since 2.402 and already has a duplicate report submitted.  I think that might give some indication that we should choose to resolve JENKINS-71156 and accept that by resolving JENKINS-71156 we are also bringing back JENKINS-71089.

Do you have a different suggestion?

**Comment by janfaracik** [comment]
> @janfaracik I see your 👍 on the comment from @NotMyFault , but I don't see a pull request that resolves [JENKINS-71156](https://issues.jenkins.io/browse/JENKINS-71156). Is there a fix in progress or should we consider reverting this change?
> 
> The report from @daniel-beck in [JENKINS-71089](https://issues.jenkins.io/browse/JENKINS-71089) did not seem to be as readily detected as [JENKINS-71156](https://issues.jenkins.io/browse/JENKINS-71156). The behavior in JENKINS-71089 has existed since 2.335, while the behavior reported in JENKINS-71156 has only been seen since 2.402 and already has a duplicate report submitted. I think that might give some indication that we should choose to resolve JENKINS-71156 and accept that by resolving JENKINS-71156 we are also bringing back JENKINS-71089.
> 
> Do you have a different suggestion?

Hey, happy to either do a fix or revert. Do we know if other hetero-lists are affected by the change in 71156 or if it's just limited to the parameterised job list?

**Comment by Bananeweizen** [comment]
@janfaracik The view configuration also doesn't work correctly ( /view/foo/configure). For each column, there is only an "empty container". This is on 2.403:
image

**Pull request merged**
